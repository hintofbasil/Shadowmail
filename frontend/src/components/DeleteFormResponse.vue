<template>
  <div class="row">
    <div :class="this.get_class_name()">
      <i
        class="material-icons descriptor-icon"
        v-text="this.get_material_icon()"
      >
      </i>
      <br />
      <span :class="this.get_class_name()">{{ this.reason }}</span>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    status: String,
    reason: String
  },
  methods: {
    get_class_name: function() {
      return this.status === "OK"
        ? "offset-by-two eight columns success-response"
        : "offset-by-two eight columns error-response";
    },
    get_material_icon: function() {
      return this.status === "OK" ? "mail_outline" : "error";
    }
  }
};
</script>

<style lang="scss" scoped>
@import "../colors";

.descriptor-icon {
  font-size: 2em;
}

.success-response {
  text-align: center;
  color: $success-text-color;
  font-size: 1.5em;
}

.error-response,
.error-response-small {
  text-align: center;
  color: $error-text-color;
  font-size: 1.5em;
}
</style>
