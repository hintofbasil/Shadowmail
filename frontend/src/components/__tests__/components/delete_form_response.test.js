import React from 'react';
import renderer from 'react-test-renderer';

import DeleteFormResponse from '../../delete_form_response';

describe('render', () => {
  it('renders successfully', () => {
    const tree = renderer
      .create(
        <DeleteFormResponse
          status="status"
          reason="reason"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
