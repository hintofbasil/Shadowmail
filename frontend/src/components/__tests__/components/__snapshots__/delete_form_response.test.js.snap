// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders successfully 1`] = `
<div
  className="row"
>
  <div
    className="offset-by-two eight columns error-response"
  >
    <i
      className="material-icons descriptor-icon"
    >
      error
    </i>
    <br />
    reason
  </div>
</div>
`;
